service/block/types.go

```diff
@@ -1,9 +1,26 @@
 package block
 
 import (
+	"github.com/celestiaorg/rsmt2d"
+
 	core "github.com/celestiaorg/celestia-core/types"
 )
 
 // Raw is an alias to a "raw" Core block. It is "raw" because
 // it is still awaiting erasure coding.
 type Raw = core.Block
+
+// ExtendedBlock contains the erasure coded block data as well as its
+// ExtendedHeader.
+type ExtendedBlock struct {
+	// header *ExtendedHeader
+	// TODO @renaynay: include pointer to ExtendedHeader
+	data *ExtendedBlockData
```

Comment:
We should spec out this type `ExtendedBlock` entirely too. We might be oversimplifying the extended block to just header + extended data (the tendermint block has some more fields in its body: https://github.com/tendermint/tendermint/blob/5ccd668c788c93dc6bc0998e4590efc1094d1b2a/types/block.go#L47-L48; I think we moved evidence into block.Data but not the last-commit).

Reply:
Hmm okay I think it would be cleaner to keep data as just the extended square and then all auxiliary fields can go into the extended header. We can maybe add additional fields to the ExtendedBlock struct if some fields are unnecessary for the ExtendedHeader and just weigh it down. Regardless would be good to get this spec'd out ASAP.

Reply:
ref: https://github.com/celestiaorg/celestia-specs/issues/204